consensus/merge/src/test/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeReorgTest.java

```diff
@@ -0,0 +1,191 @@
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class MergeReorgTest implements MergeGenesisConfigHelper {
+
+  @Mock AbstractPendingTransactionsSorter mockSorter;
+
+  private MergeCoordinator coordinator;
+
+  private static final long DIFFICULTY_LEFT = 1000;
+  private final MergeContext mergeContext = PostMergeContext.get();
+  private final ProtocolSchedule mockProtocolSchedule = getMergeProtocolSchedule();
+  private final GenesisState genesisState =
+      GenesisState.fromConfig(getGenesisConfigFile(), mockProtocolSchedule);
+
+  private final WorldStateArchive worldStateArchive = createInMemoryWorldStateArchive();
+  private final MutableBlockchain blockchain = createInMemoryBlockchain(genesisState.getBlock());
+
+  private final ProtocolContext protocolContext =
+      new ProtocolContext(blockchain, worldStateArchive, mergeContext);
+
+  private final Address coinbase = genesisAllocations().findFirst().get();
+  private final BlockHeaderTestFixture headerGenerator = new BlockHeaderTestFixture();
+  private final BaseFeeMarket feeMarket =
+      new LondonFeeMarket(0, genesisState.getBlock().getHeader().getBaseFee());
+
+  @Before
+  public void setUp() {
+    var mutable = worldStateArchive.getMutable();
+    genesisState.writeStateTo(mutable);
+    mutable.persist(null);
+
+    MergeOptions.setMergeEnabled(true);
+    this.coordinator =
+        new MergeCoordinator(
+            protocolContext,
+            mockProtocolSchedule,
+            mockSorter,
+            new MiningParameters.Builder().coinbase(coinbase).build(),
+            mock(BackwardsSyncContext.class));
+    mergeContext.setTerminalTotalDifficulty(
+        genesisState.getBlock().getHeader().getDifficulty().plus(DIFFICULTY_LEFT));
+    mergeContext.setIsPostMerge(genesisState.getBlock().getHeader().getDifficulty());
+    blockchain.observeBlockAdded(
+        blockAddedEvent ->
+            blockchain
+                .getTotalDifficultyByHash(blockAddedEvent.getBlock().getHeader().getHash())
+                .ifPresent(mergeContext::setIsPostMerge));
+  }
+
+  /* Validation scenario as described over Discord:
+  as long as a post-merge PoS block has not been finalized,
+  then you can and should be able to re-org to a different pre-TTD block
+  say there is viable TTD block A and B, then we can have a PoS chain build on A for a while
+      and then see another PoS chain build on B that has a higher fork choice weight and causes a re-org
+  once any post-merge PoS chain is finalied though, you'd never re-org any PoW blocks in the tree ever again */
+
+  @Test
+  public void reorgsAcrossTDDToDifferentTargetsWhenNotFinal() {
+    // Add N blocks to chain from genesis, where total diff is < TTD
+    List<Block> endOfWork = subChain(genesisState.getBlock().getHeader(), 10, Difficulty.of(100L));
```

Comment:
I am not sure how much difference this makes, but 10 blocks * 100 diff = 1000 total diff == TTD exactly.  I think the tenth block here is technically the terminal block.